data/addFieldsToCollection.js

```diff
@@ -88,6 +109,77 @@ async function setLevelInCourses() {
   });
 }
 
+/* 
+  Script to calculate the post reqs for each sis course.
+*/
+async function setPostReqsInSISCourses() {
+  await db.connect();
+  var counter = 0;
+  SISCV.find({ version: { $elemMatch: { postReq: { $size: 0 } } } }).then(
+    async (res) => {
+      for (let course of res) {
+        let postReqs = [];
+        await SISCV.find({
+          versions: {
+            $elemMatch: {
+              preReq: { $elemMatch: { Expression: { $regex: course.number } } },
+            },
+          },
+        }).then(async (res2) => {
+          for (let matchedCourse of res2) {
+            for (let preReq of matchedCourse.versions[0].preReq) {
+              if (preReq.IsNegative === 'N') {
+                postReqs.push({
+                  courseId: matchedCourse.id,
+                  number: matchedCourse.number,
+                  title: matchedCourse.title,
+                  credits: matchedCourse.versions[0].credits,
+                  preReqs: preReq.Expression,
+                });
+                break;
+              }
+            }
+          }
+        });
+        counter++;
+        for (let version of course.versions) {
+          version.postReq = postReqs;
+        }
+        if (counter % 10 === 0) {
+          console.log(course.title);
+        }
+        await course.save();
+      }
+    },
+  );
+  console.log('done');
+}
+
+/* 
+  Script to set post req field of each current course by looking for corresponding sis course.
+*/
+async function setPostReqsInCourses() {
+  await db.connect();
+  courses.find().then(async (res) => {
```

Comment:
What is this async chaining XD

Reply:
Nice to have: move away from .then and use async await for readability ;-; 

just a nitpick though